scripts/benchmark.cr

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env -S crystal run --no-debug --release
+require "../src/pluto"
+
+record BenchmarkResult, time : Int64, memory : Int64
+
+def benchmark_memory
+  bytes_before_measure = GC.stats.total_bytes
+  yield
+  (GC.stats.total_bytes - bytes_before_measure).to_i64
+end
+
+def benchmark_time
+  time = Time.monotonic
+  yield
+  Time.monotonic - time
+end
+
+def print_result_table(total : Hash(String, BenchmarkResult))
+  name_rjust = total.keys.map(&.size).max
+  time_ljust = total.values.map { |t| t.time.to_s.size + 2 }.max
+  memo_ljust = total.values.map(&.memory.humanize_bytes.size).max
+
+  # Headers
+  table = [
+    [" " * name_rjust, "Time".ljust(time_ljust), "Memory".ljust(memo_ljust)],
+    ["-" * name_rjust, "-" * time_ljust, "-" * memo_ljust],
+  ]
+
+  # Rows
+  total.each do |name, t|
+    table << [
+      name.rjust(name_rjust),
+      "#{t.time}ms".ljust(time_ljust),
+      t.memory.humanize_bytes.ljust(memo_ljust),
+    ]
+  end
+
+  # Output table itself
+  puts table.map { |r| "| #{r.join(" | ")} |" }.join("\n")
+
+  # Total Results
+  puts "\nTotal Time  : #{total.sum { |_, v| v.time }}ms"
+  puts "Total Memory: #{total.sum { |_, v| v.memory }.humanize_bytes}"
```

Comment:
It still seems that `String.build` might be cleaner.

E.g.:

```crystal
output = String.build do |string|
  # Perhaps rename `table` to `rows`?
  table.each do |row|
    string << row.join(" | ") << "\n"
  end
  string << "Total Time: " << total.sum { |_, value| value.time } << "ms"
  string << "Total Memory: " << total.sum { |_, value| value.memory }.humanize_bytes
end
```

Reply:
Btw, not sure what is the purpose of displaying total time and memory. When doing changes, wouldn't you look on specific operations anyway?

Reply:
I can tack on another `Total` row with the time and memory usage as just another row? I don't agree that using the `String.build` method would make this cleaner, but that might be a _me_ thing :shrug: I can change it to that regardless if you think it will improve readability. 

I've been using the total fields as a way to see if anything significant changed between runs. If the numbers don't change significantly from the previous run, I don't bother looking at the specific operations.